src/hal/src/device.rs

```diff
@@ -734,6 +734,38 @@ pub trait Device<B: Backend>: fmt::Debug + Any + Send + Sync {
         flags: query::ResultFlags,
     ) -> Result<bool, WaitError>;
 
+    /// Create an acceleration structure object.
+    unsafe fn create_acceleration_structure(
+        &self,
+        desc: &acceleration_structure::AccelerationStructureDesc<B>,
+    ) -> Result<B::AccelerationStructure, OutOfMemory>;
+
+    /// Destroy an acceleration structure object.
+    unsafe fn destroy_acceleration_structure(
+        &self,
+        acceleration_structure: B::AccelerationStructure,
+    );
+
+    /// Get the size requirements for the buffers needed to build an acceleration structure.
+    unsafe fn get_acceleration_structure_build_requirements(
+        &self,
+        // used for host operations
+        // build_type: acceleration_structure::HostOrDevice,
+        build_info: &acceleration_structure::AccelerationStructureGeometryDesc<B>,
+        // must be a parallel array to `build_info.geometries` containing the primitive counts for each geometry.
+        max_primitives_count: &[u32],
+    ) -> acceleration_structure::AccelerationStructureSizeRequirements;
+
+    // TODO these are used for host operations, which DX12 does not support.
```

Comment:
Note these functions (in vulkan) optionally take a deferral (`VK_KHR_deferred_host_operations`), which I don't think gfx has a concept in its API for..?

Reply:
correct, we don't have anything like this yet...

Reply:
AFAIK that extension is only used by acceleration structure host commands, but regardless I've made a PR in ash https://github.com/MaikKlein/ash/pull/348 so hopefully we'll be able to use it by the next release of ash